tests/explainers/test_tree.py

```diff
@@ -1197,6 +1197,30 @@ def test_xgboost_regression(self):
         expected_diff = np.abs(explanation.values.sum(1) + explanation.base_values - predicted).max()
         assert expected_diff < 1e-4, "SHAP values don't sum to model output!"
 
+    def test_xgboost_dmatrix_propagation(self):
+        xgboost = pytest.importorskip("xgboost")
+
+        def sigmoid(x):
+            return 1 / (1 + np.exp(-x))
+
+        X, y = shap.datasets.adult()
+
+        # Randomly set some features of X to NaN to test propagation of missing param
```

Comment:
The comment is a little incoherent with the code below, you're setting some values of X to 1e-8, not NaN